src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -25,7 +26,7 @@ namespace NuGet.Packaging
     public abstract class PackageReaderBase : IPackageCoreReader, IPackageContentReader, IAsyncPackageCoreReader, IAsyncPackageContentReader, ISignedPackageReader
     {
         private NuspecReader _nuspecReader;
-
+        private static readonly Regex _resourceDllRegex = new Regex(@"[\\\/](\w{2}|\w{2}[-]\w{2})[\\\/].+[.]resources[.]dll$", RegexOptions.IgnoreCase);
```

Comment:
I found [this doc](https://docs.microsoft.com/en-us/dotnet/framework/resources/packaging-and-deploying-resources-in-desktop-apps#resource-naming-conventions) which says:

> There are some exceptions for the two-letter culture names, such as zh-Hans for Chinese (Simplified).

NuGet's docs on [how to create localized NuGet packages](https://docs.microsoft.com/en-us/nuget/create-packages/creating-localized-packages#localized-resource-assemblies-in-a-single-package) uses `zh-Hans` and `zh-Hant` in the example. Looking at the results of `CultureInfo.GetCultures(CultureTypes.AllCultures).Select(c => c.Name)`, there are many that would cause this regex to return a false negative.

This regex (or the place where it's used) would need to be improved in order to avoid a change in NuGet's behaviour if a customer is creating a satellite assembly for one of these locales. I would strongly recommend a test that tests every culture returned from `CultureInfo.GetCultures(CultureTypes.AllCultures)` to reduce the risk that we regress current behaviour. In fact, I'm not sure I'll approve the PR without such a test. Although I'm not really sure what the consequence would be of a regression. Would it be a problem in pack or in consuming the package? Will it lead to build errors or warnings?

Given that satellite assemblies are always in a subfolder, perhaps another way to achieve similar results is to detect when an assembly is directly in the `lib/<tfm>/` directory, so that `lib/<tfm>/<anything>/*.resources.dll` can be skipped, even when `<anything>` is not a language code? I'll need to do an analysis of all packages on nuget.org, but I think that will have a lower risk of false negatives.

Reply:
@zivkan  Thanks for the advice. Regex is changed to check the only resources.dll under the folders "lib//<anything>/<anything>/*.resources.dll".   My bad about the cultures, new unit test is added to check all the entries like it's suggested. MSBuildNuGetProject is also covered by the unit test now.